pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -630,6 +647,28 @@ func (c *Controller) onNodeEvent(_, node *v1.Node, event model.Event) error {
 	return nil
 }
 
+func makeWEkey(entry *v1alpha3.WorkloadEntry) string {
+	return entry.Spec.Address + "/" + entry.Spec.Network
+}
+
+func (c *Controller) onWorkloadEntryEvent(_, entry *v1alpha3.WorkloadEntry, event model.Event) error {
+	// if a workloadentry is created with the same IP address & Network as an already existing workloadentry, delete the old workloadentry.
```

Comment:
I think we probably only want to do this for auto-registration cases. Deleting user-created resources is confusing and a bad UX. A user _shouldn't_ have overlapping IPs, but that should be resolve by other conflict resolution mechanism that aren't "delete the workload entry" IMO.

I also know some users _are_ doing this, which this would instantly break them I think.

Reply:
so like, check for the `istio.io/autoRegistrationGroup` label and ignore if it's not present?